### PR TITLE
WIP: Downgrade log level for access rejections as a consequence of invalid credentials/expired token

### PR DESCRIPTION
**Pull request by catper**
## Related issue

[#2031](https://github.com/ory/hydra/issues/2031)

## Proposed changes

Invalid credentials or an expired token is a normal thing to happen and should be logged on "info" level, not on "error" level as is currently the case.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Comment by aeneasr** [comment]
Looking great, thank you!

File: oauth2/handler.go [hunk 1]
```diff
@@ -587,7 +587,7 @@ func (h *Handler) TokenHandler(w http.ResponseWriter, r *http.Request) {
 
 	accessResponse, err := h.r.OAuth2Provider().NewAccessResponse(ctx, accessRequest)
 	if err != nil {
-		x.LogError(r, err, h.r.Logger())
+		x.LogAudit(r, err, h.r.Logger())
```
Comment (aeneasr) [inline oauth2/handler.go:590]: Why did you change this here? :)

Comment (catper) [reply oauth2/handler.go:590]: perhaps i'm mistaken but is this not where the error is logged in case credentials are invalid? my aim is for that error log to be downgraded to something less severe.

Comment (aeneasr) [reply oauth2/handler.go:590]: That is correct, however there could be a number of issues which would warrant a error log such as a broken database connection and more. Maybe we should check for the error type and choose the logger only for cases where invalid credentials are sent?

Comment (catper) [reply oauth2/handler.go:590]: okay, my bad, i thought invalid credentials was the only remaining error here. will fix!

File: oauth2/handler.go [hunk 2]
```diff
@@ -549,8 +549,16 @@ func (h *Handler) TokenHandler(w http.ResponseWriter, r *http.Request) {
 	var ctx = r.Context()
 
 	accessRequest, err := h.r.OAuth2Provider().NewAccessRequest(ctx, r, session)
+
 	if err != nil {
-		x.LogError(r, err, h.r.Logger())
+		switch err.Error() {
+		case fosite.ErrServerError.Error():
+		case fosite.ErrTemporarilyUnavailable.Error():
```
Comment (aeneasr) [inline oauth2/handler.go:556]: Golang does not automatically fallthrough, you need to add the fallthrough statement to do so. In general I think you could simplify this as:

```
if errors.Is(err, fosite.ErrUnauthorizedError) {
  x.LogAudit(r, err, h.r.Logger())
} else {
  x.LogError(r, err, h.r.Logger())
}
```

Comment (catper) [reply oauth2/handler.go:556]: sorry about the fallback, guess that shows how much go experience i have ...

the problem with using `ErrRequestUnauthorized` (I assume this is what you mean, there doesn't seem to be an `ErrUnauthorizedError`?) is that that's not actually what's returned, we get something more specific like `ErrInvalidClient`, `ErrUnsupportedGrantType` or similar.

Comment (aeneasr) [reply oauth2/handler.go:556]: No problem, it is a good learning experience :)

The error I meant was `fosite. ErrUnauthorizedClient` and `fosite.ErrInactiveToken`

Comment (catper) [reply oauth2/handler.go:556]: the error comparisons won't return true though. one could potentially check e.g.  `strings.Compare(err.Error(), fosite.ErrInvalidClient.Name) == 0` instead but not sure that that's the nicest solution.

regardless, i still think that anything that doesn't correspond to a status code >= 500 should be passed to the audit log rather than the errror log.

Comment (zepatrik) [reply oauth2/handler.go:556]: You would want to use `errors.Cause(err) == fosite.Err...` from https://github.com/pkg/errors#retrieving-the-cause-of-an-error

**Review by aeneasr — APPROVED** [review]
Awesome 🎉 

Thank you for your contribution!
